pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -675,24 +681,26 @@ func (s *store) createPermissions(sess *db.Session, roleID int64, resource, reso
 		permissions = append(permissions, p)
 	}
 
-	// If there are no missing actions for the resource (in case of access level downgrade or resource removal), we don't need to insert any prior actions
+	// If there are no missing actions for the resource (in case of access level downgrade or resource removal), we don't need to insert any actions
 	// we still want to add the action set in case of access level downgrade, but not in case of resource removal (when permission == "")
-	if len(missingActions) == 0 {
-		if s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) && permission != "" {
-			if _, err := sess.InsertMulti(&permissions); err != nil {
-				return err
-			}
+	if len(missingActions) == 0 && permission != "" && s.features.IsEnabled(context.TODO(), featuremgmt.FlagAccessActionSets) {
+		if _, err := sess.InsertMulti(&permissions); err != nil {
+			return err
 		}
 		return nil
 	}
```

Comment:
I think we still want to return here if `len(missingActions) == 0` is true, but the other conditions are not.